views/js/review/plugins/navigation/review-panel/panel-data.js

```diff
@@ -94,36 +96,92 @@ define([
         return data;
     };
 
+    /**
+    * Refines the test runner data and build the expected review panel map
+    * @param {testMap} testMap
+    * @param {Boolean} withScore
+    * @returns {reviewPanelMap}
+    */
+    const getAccordionReviewPanelMap = (testMap, withScore = true) => {
+        const { parts, score, maxScore } = testMap;
+        const items = new Map();
+
+        // rebuild the map keeping only relevant data, and sorting elements by position
+        const panelMap = {
+            parts: _.map(parts, part => Object.assign(extractData(part, withScore), {
+                sections: _.map(part.sections, section => Object.assign(extractData(section, withScore), {
+                    items: _.map(section.items, item => {
+                        const reviewItem = extractData(item, withScore);
+                        reviewItem.type = getItemType(item, withScore);
+                        items.set(item.id, reviewItem);
+                        return reviewItem;
+                    }).sort(compareByPosition)
+                })).sort(compareByPosition)
+            })).sort(compareByPosition),
+            withScore,
+            items
+        };
+
+        if (withScore) {
+            Object.assign(panelMap, { score, maxScore });
+        }
+
+        return panelMap;
+    };
+
     return {
+        getAccordionReviewPanelMap,
+
         /**
          * Refines the test runner data and build the expected review panel map
          * @param {testMap} testMap
          * @param {Boolean} withScore
+         * @param {Boolean} displaySectionTitles
          * @returns {reviewPanelMap}
          */
-        getReviewPanelMap(testMap, withScore = true) {
-            const {parts, score, maxScore} = testMap;
-            const items = new Map();
-
-            // rebuild the map keeping only relevant data, and sorting elements by position
-            const panelMap = {
-                parts: _.map(parts, part => Object.assign(extractData(part, withScore), {
-                    sections: _.map(part.sections, section => Object.assign(extractData(section, withScore), {
-                        items: _.map(section.items, item => {
-                            const reviewItem = extractData(item, withScore);
-                            reviewItem.type = getItemType(item, withScore);
-                            items.set(item.id, reviewItem);
-                            return reviewItem;
-                        }).sort(compareByPosition)
-                    })).sort(compareByPosition)
-                })).sort(compareByPosition),
-                withScore,
-                items
-            };
-
-            if (withScore) {
-                Object.assign(panelMap, {score, maxScore});
-            }
+        getFizzyReviewPanelMap(testMap, withScore = true, showCorrect = false) {
```

Comment:
At this point I don't really like having `getAccordionReviewPanelMap` and `getFizzyReviewPanelMap` which return testmaps with different item properties. I'm wondering if we can join them back together in a unified function, extend the item data as needed, and let the presentational components decide which properties they'll use.

Reply:
Yes, I see the differences are deeper than I initially thought. I would need to keep the 2 `get___ReviewPanelMap` functions (to return `parts` or `sections`), but move the item properties setting logic into a separate function (like `extractData()`). It would result in a unified item data format like that:

```
/**
 * @typedef {Object} ReviewItem
 * @property {String} id - item id // both panels
 * @property {Number} position - recommended to use 0-based list index // old
 * @property {String} label - displayed text // old
 * @property {String} numericLabel - displayed number (alternative to label) // new
 * @property {String} ariaLabel // new
 * @property {Number} score - the item's current score // old
 * @property {Number} maxScore - the item's max possible score // old
 * @property {Boolean} informational // old
 * @property {Boolean} skipped // old
 * @property {String} type - 'correct'/'incorrect'/'info'/'skipped'/'default' // old
 * @property {String} status - 'answered'/'viewed'/'unseen' // new
 * @property {String} scoreType - 'correct'/'incorrect'/null // new
 * @property {String} icon - 'info' or null - replaces label and numericLabel if set // new
 */
```

But it's a messy result and may not be worth the effort of refactoring. @jsconan do you have any preference?

Reply:
**thought:** Well, I'd rather go to a unified version too. Even if it may look messy as it will contain more data, if the only difference is having additional entries, I don't think it is so bad. If the whole structure is different, however, it might require different functions, but this is only about part vs section right? so at the end the structure remains the same, but the scope is different.

**suggestion:** Why not having only one `getReviewPanelMap`, with a scope parameter to reduce the amount f data (section, test part, etc.)

Reply:
I've refactored this back into the single `getReviewPanelMap` function with the above item data structure.

Reply:
Note: I'm fine too keep it like now, but logically I think it's **not** a unified structure. You rely on the fact that `itemButtonList` and `accordionList` components won't have conflicts in property names - but why should they?
But it works for our case, so let's keep it.